crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net_task.rs

```diff
@@ -582,6 +266,366 @@ impl MetaNetTask {
+                                return Err(MetaNetTaskError::RequiredChannelClosed)
+                            }
+                            _ => {
+                                // Ignore other errors
+                            }
+                        }
+                    }
+
+                    for (coord, bound) in regions {
+                        match self
+                            .evt_sender
+                            .fetch_op_data(FetchOpDataEvt {
+                                space: space.clone(),
+                                query: FetchOpDataEvtQuery::Regions(vec![bound]),
+                            })
+                            .await
+                        {
+                            Ok(list) => {
+                                let last_idx = list.len() - 1;
+                                for (idx, (_hash, op)) in list.into_iter().enumerate() {
+                                    self.fetch_response_queue.enqueue_op(
+                                        space.clone(),
+                                        (con.clone(), url.clone(), Some((coord, idx == last_idx))),
+                                        op,
+                                    );
+                                }
+                            }
+                            Err(KitsuneP2pError::GhostError(GhostError::Disconnected)) => {
+                                return Err(MetaNetTaskError::RequiredChannelClosed)
+                            }
+                            _ => {
+                                // Ignore other errors
+                            }
+                        }
+                    }
+                }
+
+                Ok(())
+            }
+            wire::Wire::PushOpData(wire::PushOpData { op_data_list }) => {
+                for (space, op_list) in op_data_list {
+                    for op in op_list {
+                        // hash the op
+                        let op_hash = match self.host.op_hash(op.op_data.clone()).await {
+                            Ok(op_hash) => op_hash,
+                            Err(err) => {
+                                tracing::warn!(
+                                    ?err,
+                                    "Dropping incoming op because the host failed to hash it {:?}",
+                                    op
+                                );
+                                continue;
+                            }
+                        };
+
+                        // MAYBE: do something with the
+                        //        is_last bool?
+                        //        Right now we don't
+                        //        really care, because
+                        //        if it's a region
+                        //        we know it's gossip
+                        //        so it's okay if
+                        //        the context is
+                        //        `None`.
```

Comment:
```suggestion
                        // MAYBE: do something with the is_last bool?
                        //        Right now we don't really care, because
                        //        if it's a region we know it's gossip
                        //        so it's okay if the context is `None`.
```